synapse/federation/federation_server.py

```diff
@@ -1521,6 +1525,21 @@ def _get_event_ids_for_partial_state_join(
     if current_membership_event_id is not None:
         state_event_ids.add(current_membership_event_id)
 
+    name_id = prev_state_ids.get((EventTypes.Name, ""))
+    canonical_alias_id = prev_state_ids.get((EventTypes.CanonicalAlias, ""))
+    if not name_id and not canonical_alias_id:
+        # Also include the hero members of the room (for DM rooms without a title).
+        # To do this properly, we should select the correct subset of membership events
+        # from `prev_state_ids`. Instead, we are lazier and use the (cached)
+        # `get_room_summary` function, which is based on the current state of the room.
+        # This introduces races; we choose to ignore them because a) they should be rare
+        # and b) even if it's wrong, joining servers will get the full state eventually.
+        heroes = extract_heroes_from_room_summary(summary, join_event.state_key)
+        for hero in heroes:
+            membership_event_id = prev_state_ids.get((EventTypes.Member, hero))
+            if membership_event_id:
+                state_event_ids.add(membership_event_id)
+
     # TODO: return a few more members:
     #   - those with invites
     #   - those that are kicked? / banned
```

Comment:
Unclear if this TODO still stands?

Reply:
I guess it's not related to your work here, though ideally this should be an issue so we can talk about it. I'm not convinced the TODO is a good idea.

Reply:
oh, I see, we think this comment might've just been talking about the heroes?
Yeah I'd be tempted to knock the comment out.

Reply:
> Yeah I'd be tempted to knock the comment out. 

have done so.